### PR TITLE
updates the expandable component

### PR DESCRIPTION
**Pull request by mirnawong1**
this pr iterates on the expandable component to improve the user experience based on some feedback [slack thread](https://dbt-labs.slack.com/archives/C02NCQ9483C/p1714405513625989):

- separated the expandable component from the headerLinkCopy.js file as i think that was conflicting with the behavior.
- allows copying and text highlighting inside the expandable
- added separate code for a copy link pop up/close behavior. `showCopyPopup` and `closeButton` with timeout.
- added css info so expandable alt_header text is black when using details theme. 
- also adds a `useEffect` behavior so that the toggle is expanded when the user goes to the copied link so they know where to start reading

[asana task](https://app.asana.com/0/1200099998847559/1207358280519061/f) requesting PR review

### CODE REVIEW

File: website/src/components/expandable/styles.module.css [hunk 1]
```diff
@@ -4,14 +4,49 @@
     text-decoration: none;
     transition: text-decoration 0.3s; /* Smooth transition */
     font-weight: 600;
-    margin-bottom: 20px;
+    margin-bottom: 10px;
```
Comment (mirnawong1) [inline website/src/components/expandable/styles.module.css:7]: fixed margin bc spacing was too wide in between expandables.

File: website/src/components/expandable/index.js [hunk 2]
```diff
@@ -1,47 +1,84 @@
...
-    .replace(/\-\-+/g, '-') // Replace multiple - with single -
-    .replace(/^-+/, '') // Trim - from start
-    .replace(/-+$/, ''); // Trim - from end
+    .normalize('NFD')
+    .replace(/[\u0300-\u036f]/g, '')
+    .replace(/\s+/g, '-')
+    .replace(/[^\w\-]+/g, '')
+    .replace(/\-\-+/g, '-')
+    .replace(/^-+/, '')
+    .replace(/-+$/, '');
```
Comment (john-rock) [inline website/src/components/expandable/index.js]: I'm in the camp of leaving the comments here, especially for chained regex, but not considering that as a blocker!

File: website/src/components/expandable/index.js [hunk 3]
```diff
@@ -1,47 +1,84 @@
...
+
+useEffect(() => {
+  if (window.location.hash === `#${anchorId}`) {
+    setOn(true);
+    const element = document.getElementById(anchorId);
+    if (element) {
+      element.scrollIntoView({ behavior: 'smooth' });
+    }
+  }
+}, [anchorId]);
```
Comment (john-rock) [inline website/src/components/expandable/index.js:69]: Nice 😎

File: website/src/components/expandable/index.js [hunk 4]
```diff
@@ -1,47 +1,84 @@
...
+    .replace(/\s+/g, '-')
+    .replace(/[^\w\-]+/g, '')
+    .replace(/\-\-+/g, '-')
+    .replace(/^-+/, '')
+    .replace(/-+$/, '');
 }
 
 function expandable({ children, alt_header = null }) {
-  if(!alt_header) { return null; }
+  if (!alt_header) { return null; }
```
Comment (john-rock) [inline website/src/components/expandable/index.js]: We should capitalize Expandable here to follow React component naming conventions

React components are regular JavaScript functions except:
[Their names always begin with a capital letter.](https://react.dev/learn/your-first-component#recap)

**Review by john-rock — CHANGES_REQUESTED** [review]
Hey @mirnawong1 nice work!

I only have one small request to adjust the case of the component to follow React's standards.

File: website/src/components/expandable/index.js (hunk 4 above)
Comment (mirnawong1) [reply website/src/components/expandable/index.js]: oh thanks for sharing that -  that's great to know and sorry my previous components will have the bad practice 🙈

**Comment by mirnawong1** [comment]
hey @john-rock , thanks for that great feedback! I've folded in your changes and can you rereview when you have chance ? 🙏

File: website/src/components/expandable/index.js (hunk 4 above)
Comment (john-rock) [reply website/src/components/expandable/index.js]: No worries!

**Review by john-rock — APPROVED** [review]
lgtm @mirnawong1 , nice work on this :shipit:

**Comment by mirnawong1** [comment]
thank you so much @john-rock !

**Comment by mirnawong1** [comment]
@john-rock , i think i need to revert this pr: https://docs.getdbt.com/docs/collaborate/dbt-explorer-faqs the faqs aren't appearing correctly

**Pull request merged**